Sources/TokamakLint/TokenVisitor.swift

```diff
@@ -0,0 +1,234 @@
+    if syntax.hasSuffix("Syntax") {
+      syntax = String(syntax.dropLast(6))
+    }
+
+    let node = Node(text: syntax)
+    node.range.startRow = row
+    node.range.startColumn = column
+    node.range.endRow = row
+    node.range.endColumn = column
+    if current == nil {
+      tree.append(node)
+    } else {
+      current.add(node: node)
+    }
+    current = node
+  }
+
+  override func visit(_ token: TokenSyntax) -> SyntaxVisitorContinueKind {
+    current.text = escapeHtmlSpecialCharacters(token.text)
+    current.token = Node.Token(
+      kind: "\(token.tokenKind)",
+      leadingTrivia: "",
+      trailingTrivia: ""
+    )
+
+    current.range.startRow = row
+    current.range.startColumn = column
+
+    token.leadingTrivia.forEach { piece in
+      let trivia = processTriviaPiece(piece)
+      current.token?.leadingTrivia += replaceSymbols(text: trivia)
+    }
+    processToken(token)
+    token.trailingTrivia.forEach { piece in
+      let trivia = processTriviaPiece(piece)
+      current.token?.trailingTrivia += replaceSymbols(text: trivia)
+    }
+
+    current.range.endRow = row
+    current.range.endColumn = column
+
+    return .visitChildren
+  }
+
+  override func visitPost(_ node: Syntax) {
+    current.range.endRow = row
+    current.range.endColumn = column
+    current = current.parent
+  }
+
+  private func processToken(_ token: TokenSyntax) {
+    var kind = "\(token.tokenKind)"
+    if let index = kind.firstIndex(of: "(") {
+      kind = String(kind.prefix(upTo: index))
+    }
+    if kind.hasSuffix("Keyword") {
+      kind = "keyword"
+    }
+
+    column += token.text.count
+  }
+
+  private func processTriviaPiece(_ piece: TriviaPiece) -> String {
+    var trivia = ""
+    switch piece {
+    case let .spaces(count):
+      trivia += String(repeating: "&nbsp;", count: count)
+      column += count
+    case let .tabs(count):
+      trivia += String(repeating: "&nbsp;", count: count * 2)
+      column += count * 2
+    case let .newlines(count),
+         let .carriageReturns(count),
+         let .carriageReturnLineFeeds(count):
+      trivia += String(repeating: "<br>\n", count: count)
+      row += count
+      column = 0
+    case let .backticks(count):
+      trivia += String(repeating: "`", count: count)
+      column += count
+    case let .lineComment(text):
+      trivia += withSpanTag(class: "lineComment", text: text)
+      processComment(text: text)
+    case let .blockComment(text):
+      trivia += withSpanTag(class: "blockComment", text: text)
+      processComment(text: text)
+    case let .docLineComment(text):
+      trivia += withSpanTag(class: "docLineComment", text: text)
+      processComment(text: text)
+    case let .docBlockComment(text):
+      trivia += withSpanTag(class: "docBlockComment", text: text)
+      processComment(text: text)
+    default:
+      break
+    }
+    return trivia
+  }
+
+  private func withSpanTag(class c: String, text: String) -> String {
```

Comment:
same request for this function and similar functions below: their purpose is unclear, and there has to be comment that explains that in detailed